### PR TITLE
Release 1.2.0 - Support Adams + Merge develop into main 

### PR DESCRIPTION
**Pull request by aljidy**
## Description

- Includes work for #71  - Support Adams booking 
- Bump version to 1.2.0, version code and project version 
- merge develop into main, assuming it was missed after last release

## Things to check

Checklist:

- [x] Two reviews
- [x] Manually tested iOS
- [x] Manually tested Android
- [ ] (Optional) Updated the [Confluence docs](https://anddigitaltransformation.atlassian.net/wiki/spaces/ML/pages/4398940396/MCDBA+Mobile+App)

### CODE REVIEW

**Review by Jake-Sturgeon — APPROVED** [review]

File: functions/src/v1/Services/Scheduled/callCarAPI.ts [hunk 1]
```diff
@@ -32,9 +32,10 @@ export async function callCarAPI() {
       'Content-Type': 'application/json',
     },
     body: JSON.stringify({date: midnightDate, spaceType: 'car'}),
-    // retry up to 3 times with backoff if 429 status code
+    // retry up to 3 times with backoff if >400 status code
     retries: 3,
-    retryOn: [429],
+    retryOn: async (_attempt, error, response) =>
+      error !== null || (response !== null && response.status >= 400),
```
Comment (tomdownes14) [inline functions/src/v1/Services/Scheduled/callCarAPI.ts:38]: Why make this broader?

I think 429 was specifically for when we get cold start issues

File: .github/workflows/distribute-apps.yml [hunk 2]
```diff
@@ -54,7 +68,7 @@ jobs:
 
       - name: Retrieve the secret and decode it to a file
         env:
-          ADMIN_SDK_BASE64: ${{ secrets.MURRAY_APPS_DEV_FIREBASE_ADMIN_BASE_64_ENCODED_JSON }}
+          ADMIN_SDK_BASE64: ${{ secrets.FIREBASE_ADMIN_BASE_64_ENCODED_JSON }}
```
Comment (tomdownes14) [inline .github/workflows/distribute-apps.yml:71]: Have the secrets been renamed and then put into environments?

File: fastlane/Fastfile
Comment (tomdownes14) [inline fastlane/Fastfile:1]: Can you explain the enterprise > adhoc change please

File: fastlane/Fastfile [hunk 3]
```diff
@@ -23,72 +22,73 @@ platform :ios do
...
+            distribute_external: true,
+            notify_external_testers: true,
+            expire_previous_builds: true,
+            groups: ["Murray App Beta Testers"],
+            beta_app_review_info: {
+                contact_email: "christopher.batin@and.digital",
+                contact_first_name: "Christopher",
+                contact_last_name: "Batin",
+                demo_account_name: "demo@example.com",
+                demo_account_password: "N^LhU%h4NV&z",
```
Comment (tomdownes14) [inline fastlane/Fastfile:88]: I wonder if this should also be a secret - though I appreciate it wasn't before now.

File: fastlane/Fastfile
Comment (Jake-Sturgeon) [reply fastlane/Fastfile:1]: I think it's to do with us not having an enterprise Apple account

**Review by Jake-Sturgeon — APPROVED** [review]

File: functions/src/v1/Services/Scheduled/callCarAPI.ts (hunk 1 above)
Comment (aljidy) [reply functions/src/v1/Services/Scheduled/callCarAPI.ts:38]: This was your change 🤣 
https://github.com/and-murray/wharf-spaces/commit/3fd3c020e0d28ae4dc76f5a71862665c40466e68

File: .github/workflows/distribute-apps.yml (hunk 2 above)
Comment (aljidy) [reply .github/workflows/distribute-apps.yml:71]: Looks like it's there 
image

File: fastlane/Fastfile
Comment (aljidy) [reply fastlane/Fastfile:1]: Yes this ^

Comment (aljidy) [reply fastlane/Fastfile:1]: If you look in the #apple-admins channel, you'll see discusion about it

**Review by tomdownes14 — APPROVED** [review]

**Pull request merged**